### PR TITLE
feat: convert thirdparty donation tests to Playwright

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Comment by Sembauke** [comment]
Alright, the tests should be good to go. It will fail in the Ci because secrets do not work in PR's

**Review by huyenltnguyen — APPROVED** [review]

File: e2e/third-party-donation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+  // test('It should have a Paypal and Patreon button', async ({ page }) => {
+  //   await page.getByRole('button', { name: 'Donate', exact: true }).click();
+  //   await expect(
+  //     page.locator('[data-paypal-smart-button-version="5.0.440"]')
+  //   ).toBeVisible();
+  //   await expect(
+  //     page.locator('[data-patreon-widget-type="become-patron-button"]')
+  //   ).toBeVisible();
+  // });
```
Comment (raisedadead) [inline e2e/third-party-donation.spec.ts:79]: Can this part be uncommented and included in tests?

Can we not test the presence of the PayPal and Patreon buttons without clicking them for the actual API call to those services? 

That is, the user story for the test will be: As I user I would like to select an option for the vendor (but not proceed further by clicking on it)

Does that makes sense? If yes, then can you do the same for the Stripe stuff below as well?

Comment (Sembauke) [reply e2e/third-party-donation.spec.ts:79]: The buttons are not present without the API Keys

Comment (Sembauke) [reply e2e/third-party-donation.spec.ts:79]: Meaning if we want to test this we would include the secrets in the Playwright workflow or a separate workflow that only runs on the staging and/or production branch.

**Comment by ahmaxed** [comment]
Thank you for your patience, @Sembauke. I had a chance to look into testing payment elements' (paypal button and stripe elements) best practices.

To test the functionality, all papal and stripe calls should be intercepted and replaced with mock responses for different scenarios. That way we avoid the issues associated testing against external apis.

To test if the payment elements are displayed properly, there are two routes and both have their pros and cons:
- completely mock the payment elements iframes. 
- pass only the client keys to load the actual elements.

I personally think the hybrid solution of testing actual elements and intercepted api calls would be the way to go, but let's hear from others before proceeding.

**Comment by ojeytonwilliams** [comment]
> I personally think the hybrid solution of testing actual elements and intercepted api calls would be the way to go, but let's hear from others before proceeding.

Thanks for looking into this, Ahmad. I agree that this strikes the best balance. It makes sure that we've done our part correctly without the issue inherent to testing a third party. Our part being using the PayPal and Stripe libraries correctly in learn.

File: e2e/third-party-donation.spec.ts (hunk 1 above)
Comment (raisedadead) [reply e2e/third-party-donation.spec.ts]: Ahmad says, we only need client-side keys for that. Client-side keys are not sensitive and are part of our client app - shipped to browsers.

**Review by raisedadead — CHANGES_REQUESTED** [review]
See prev. review.

**Comment by Sembauke** [comment]
@raisedadead this should do it.

**Comment by ahmaxed** [comment]
Here are further considerations:
- our api is currently mocked. To have an e2e, the calls from our api to stripe should be mocked instead.
- since we are not passing the secrets, this test could run with other e2e test in the same workflow.
- if that is the case, we can have these tests as a part of the donate page tests and remove the e2e third party workflow.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**